packages/block-library/src/image/edit.js

```diff
@@ -168,7 +168,7 @@ class ImageEdit extends Component {
 		if ( value === LINK_DESTINATION_NONE ) {
 			href = undefined;
 		} else if ( value === LINK_DESTINATION_MEDIA ) {
-			href = this.props.attributes.url;
+			href = this.props.image.source_url;
```

Comment:
Hi @earnjam thank you for your contribution. I did some tests and I think we created a regression. On images inserted by URL (e.g: external images), link destination media now does not work. This happens because these images don't have  source_url. I think we can solve the problem by using this.props.image.source_url || this.props.attributes.url to fall back to the image URL when source_url does not exist.

Reply:
Ah, good catch! Didn't think about external images.